src/Core/Content/Product/Cms/ProductListingCmsElementResolver.php

```diff
@@ -92,58 +96,94 @@ private function getNavigationId(Request $request, SalesChannelContext $salesCha
         return $salesChannelContext->getSalesChannel()->getNavigationCategoryId();
     }
 
+    private function getAllActiveSortingOptions(SalesChannelContext $context): ProductSortingCollection
+    {
+        $criteria = new Criteria();
+        $criteria
+            ->addFilter(new EqualsFilter('active', true))
+            ->addSorting(new FieldSorting('priority', 'DESC'));
+
+        /** @var ProductSortingCollection $collection */
```

Comment:
nope! please annotate the repository correctly, and this should not be necessary